File: server/entity/src/school_settings.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "school_settings")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: i32,
    pub school_code: String,
    pub school_name: Option<String>,
    pub school_region: Option<String>,
    pub school_division: Option<String>,
    pub school_year: Option<String>,
    pub updated_at: chrono::NaiveDateTime,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}

File: client/lib/domain/setup/entities/school_settings.dart
import 'package:equatable/equatable.dart';

class SchoolSettings extends Equatable {
  final String id;
  final String schoolName;
  final String schoolRegion;
  final String schoolDivision;
  final String schoolYear;
  final String schoolCode;
  final String? schoolDistrict;

  const SchoolSettings({
    required this.id,
    required this.schoolName,
    required this.schoolRegion,
    required this.schoolDivision,
    required this.schoolYear,
    required this.schoolCode,
    this.schoolDistrict,
  });

  @override
  List<Object?> get props => [
        id,
        schoolName,
        schoolRegion,
        schoolDivision,
        schoolYear,
        schoolCode,
        schoolDistrict,
      ];
}
